medicines/doc-index-updater/src/delete_manager/mod.rs

```diff
@@ -181,15 +191,31 @@ mod test {
     }
 
     #[test]
-    fn test_an_error_removes_delete_message() {
-        let state_manager = TestJobStatusClient {};
+    fn not_found_error_during_delete_removes_message_since_no_need_to_retry() {
+        let state_manager = given_a_state_manager();
+        let mut removeable_message = given_we_have_a_delete_message();
+        let error = given_an_error_has_occurred(anyhow!(
```

Comment:
I'd create a method called `given_the_document_was_not_found` which returns the error we'd expect in that scenario